src/Microsoft.ML.Data/Transforms/DropSlotsTransform.cs

```diff
@@ -190,98 +240,81 @@ private static VersionInfo GetVersionInfo()
                 verReadableCur: 0x00010001,
                 verWeCanReadBack: 0x00010001,
                 loaderSignature: LoaderSignature,
-                loaderAssemblyName: typeof(DropSlotsTransform).Assembly.FullName);
+                loaderAssemblyName: typeof(SlotsDroppingTransformer).Assembly.FullName);
         }
 
-        private const string RegistrationName = "DropSlots";
-
         /// <summary>
-        /// Extra information for each column (in addition to ColumnInfo).
+        /// Initializes a new <see cref="SlotsDroppingTransformer"/> object.
         /// </summary>
-        private sealed class ColInfoEx
+        /// <param name="env">The environment to use.</param>
+        /// <param name="input">Name of the input column.</param>
+        /// <param name="output">Name of the column resulting from the transformation of <paramref name="input"/>. Null means <paramref name="input"/> is replaced. </param>
+        /// <param name="min">Specifies the lower bound of the range of slots to be dropped. The lower bound is inclusive. </param>
+        /// <param name="max">Specifies the upper bound of the range of slots to be dropped. The upper bound is exclusive.</param>
+        public SlotsDroppingTransformer(IHostEnvironment env, string input, string output = null, int min = default, int? max = null)
```

Comment:
>public [](start = 8, length = 6)

aren't we supposed to make the constructors private/internal ?

Reply:
For trainable transformers yes, but for non trainable ones, we should have public constructors. 

---
In reply to: [235557930](https://github.com/dotnet/machinelearning/pull/1683#discussion_r235557930) [](ancestors = 235557930)